soc: nordic_nrf: Enable the TF-M NS storage partition for nordic boards

joerchan opened the pull request:
Enable the TF-M NS storage partition for nordic boards. This partition is otherwise not used, and configure as secure.

This can be tested with:
```
west build -b nrf5340dk_nrf5340_cpunet samples/bluetooth/hci_ipc
west flash --erase
west build -b nrf5340dk_nrf5340_cpuapp_ns samples/bluetooth/peripheral
west flash
```
Before this change this crashes with a SecureFault.

Fixes: https://github.com/zephyrproject-rtos/zephyr/issues/59376

Update samples and test to use the storage_partition instead of bootloader secondary slot used BL2 and mcuboot.

joerchan committed:
soc: nordic_nrf: Enable the TF-M NS storage partition for nordic boards

Enable the TF-M NS storage partition for nordic boards.
This partition is otherwise not used, and configured as secure.

Fixes: #59376

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

SebastianBoe commented on soc/arm/nordic_nrf/CMakeLists.txt:
```diff
@@ -25,4 +25,8 @@ if(CONFIG_BUILD_WITH_TFM)
+
+  set_property(TARGET zephyr_property_target
+    APPEND PROPERTY TFM_CMAKE_OPTIONS -DNRF_NS_STORAGE=${CONFIG_TFM_NRF_NS_STORAGE_PARTITION}
+    )
```
Do we have a convention for translating vanilla TF-M CMake options to Kconfig?

Perhaps we should have this code next to the other CMake code that does the same thing.

And also follow the naming convention.

E.g. if the TF-M option ends with STORAGE, then I assume the Kconfig option should not end with PARTITION?

joerchan committed:
samples: soc_flash_nrf: Stop erasing outside of test partition

Stop erasing outside of the test partition.
We shouldn't really on this being accesible memory.
This region may be consider as secure when application is non-secure.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan committed:
samples: soc_flash_nrf: Print finished message

Print a finished message, so that it is clear that this in the end
of the sample. This makes it certain that we didn't crash without
output on the last test.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan committed:
samples: soc_flash_nrf: Make sure that the flash partition is valid

Make sure that the flash partition is valid.
With TF-M enabled the storage_partition can be used as memory that is
known to be configured as non-secure flash region.
The slot1_ns_partition partition is only correct when TF-M is built
with BL2 enabled.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan committed:
tests: flash: Use a flash partition that is known to be nonsecure

With TF-M enabled the storage_partition can be used as memory that is
known to be configured as non-secure flash region.
The slot1_ns_partition partition is only correct when TF-M is built
with BL2 enabled.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>
